### PR TITLE
Extract Reddit comments from a subreddit

### PR DESCRIPTION
**Pull request by rk145lr**
- Changes include the code for performing a GET request to obtain the list of available pages for a subreddit. Additional GET requests are performed for each page. The comments are obtained and placed into a list for each page. Overall a list of dictionaries is created with the keys `article_domain`, `article_url`, `comments`, `subreddit_url`, `title`.
- Unit tests have been added for the code testing the core functionality as well as edge cases.
- The GET requests use OAuth. The required environment variables are specified in the `README.md.`

### CODE REVIEW

**Review by isaac-vb — APPROVED** [review]
Looks great to me

File: public_sentiment_pipeline/extract.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    pages_list = []
+    for page in reddit_json["data"]["children"]:
+        try:
+            page_dict = {}
+            page_dict[REDDIT_TITLE_KEY] = page["data"]["title"]
+            page_dict[REDDIT_SUBREDDIT_URL] = page["data"]["permalink"]
+            page_dict[REDDIT_ARTICLE_URL] = page["data"]["url"]
+            page_dict[REDDIT_ARTICLE_DOMAIN] = page["data"]["domain"]
+            pages_list.append(page_dict)
+        except KeyError:
```
Comment (ramiz76) [inline public_sentiment_pipeline/extract.py:66]: Nice error handling!

**Review by ramiz76 — APPROVED** [review]
